### DISCUSSION TITLE
DrawingVisual performance issues

### DISCUSSION THREAD

**Discussion by lindexi** [dotnet/wpf#5547] [General]
I use dotTrace tool to test my application and find the performance issues as the image:

![](https://user-images.githubusercontent.com/16054566/137309030-d79f7940-ce2f-452b-a7c0-59aa78faa556.png)

And sorry I can not upload the code.

**Comment by omariom** [comment]
If it's FrugalObjectList's indexer, [this how it could be optimized](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4a9BrIAW2KGIAy2YO258BG6jQDaAKV4YA4jC4zlACgwBPMTAgAZh68PACUoQC6mqQMAGJQHADm2JIA8sAAVjBgGIa8uBgAPAAqAHw0AN40DDUMYlC8CtgYMHEJyZJ5BQBCeDAlpQwA+pL5GLIY0DBWtQzVtQD0CwypDYkhKfM1xADMDMUraxuSwTwMIQAmMAihWwxV1LOzvAEMHh5cHJKSDACEALzDUYFCZTUIMABkELepyuCAYhSBY1BsGEEHM4Khb0u1wYpUBdHCt0eTxqD1JpOIAHYkSDJqiAKI8KDeACCGFh11CMwpAF87rMMDooBAAO4MNzi1lQRIcAQ8VIcDCpAJsbBcRIwBkIFRiPgQLgfbACQKcm7cu78kk1O5LFZ6/i8ABeMAudxsAFkYEKIBcAJLiE5en3+wOpB0G3AsVmJRKwXC4RowP1cUZcEKJSJ3Xb7e0WZ2u04Yc5cOHE2bkilNKAMYHjemtQEjZENnmkgW1F5vOsoxuAz7fBiEQhvDghDChHHw/Gj8ehHsNtEYjtkldPYrCsXS2XyjApuGK5Wq9Wa7W6/WG7mzNdWilza1Pam1ltTFhMjAs9lmi0P2+1W3LOG+YuhcDCio4OgMFcShQBgHApM6zS8AaDABNADBCq0sgZtIfotPgXTFvwYjSLuSEGu6wY6L6AYkR4VE0WGEZcFGMZxqoiYKMmqYhBmWYPjmBxAY6IEACIwDB4w4cm+GEUWJZlnclaktWz50lMDBNguUxtk8a5dh42mwJp/ZfD8w6zmEU54oCHhjmERmCCIy4PhWa6zBuIqitucquHupbXIeKpqhqWo6jAzEeFe/6ubUf7trFNQGY55y4No0l4TABFjAMDCJhquH4eWd61E++WJIVWUsIYBpasy3i6Xea5Po5b71V+U4/rM8X3rMnretRoZ0QxQ1pMxrGxvGnHcWmfFRAJexCQ6+AFhc4mSdKUDYN4mXZQU8mdUpa6qSlWkvrAjUxXeyXnX2EpmUOI52XO1kzs9Dm3UuYRrspJWeVuMq+Tw+6BUqwUnmF57IZeajXolDA9XD12vIZt2pQwm3bbthG5foW07bJYzFSVTA0njWOEwUbUfmyHKdZdCV3i1n3vp+dMBTcuk9Xc9SNM0rTMEgTAoPsm7eYDu4gwgQXHqFZ4RReUVHfDQpeRKMBShLfkyyFp7hZFABEU4G11cU0FaNA0OOMhcCkDAmAUW05EwMTxEkKSEb0uD9GUlQ8w0TQtCWxaQOYunZns44MM5Zz/IMoc8OHC328AjvYM7Bys7TB0cz+FvUNbUC2z83spK6LvpQVMlZTjZQMCAbTu50Yxez75TUL9vOB60BwjLVWdJ7MgkMDVbhZ5pgx92P9WD6VewQFxUANFcuZZx1ZyHQ+v1DzSU91TTXPm5bBc8Dbdul9IoFkBjS8UzXOV1w3bsdJ7fQDH7D5d/z+w2BEwx+Q0VQs9tjz0XsvHuDA17syIrnZWTNd4AN4KoGwU4IiH2oLyIAA==):

**Comment by vishalmsft** [comment]
Hey @lindexi  can you please confirm if you are looking for some performance benefit?

**Comment by lindexi** [comment]
@vishalmsft  Yes. I am thinking about some new API.

**Comment by lindexi** [comment]
[DrawingContext. Close - low performance](https://social.msdn.microsoft.com/Forums/vstudio/en-US/69d5ea92-cbd0-44b2-ad03-b7d3b8811b1e/drawingcontext-close-low-performance?forum=wpf )
